superbench/benchmarks/model_benchmarks/model_base.py

```diff
@@ -376,22 +395,26 @@ def __process_model_result(self, model_action, precision, step_times):
         precision_metric = {'float16': 'fp16', 'float32': 'fp32', 'float64': 'fp64', 'bfloat16': 'bf16'}
         if precision.value in precision_metric.keys():
             precision = precision_metric[precision.value]
-        metric = '{}_{}_step_time'.format(precision, model_action)
-        reduce_type = ReduceType.MAX if model_action is ModelAction.TRAIN else None
-        self._result.add_raw_data(metric, step_times)
-        self._result.add_result(metric, statistics.mean(step_times), reduce_type=reduce_type)
-        if model_action == ModelAction.INFERENCE:
-            self._process_percentile_result(metric, step_times, reduce_type=reduce_type)
-
+        metric_s = '{}_{}_step_time'.format(precision, model_action)
+        metric_t = '{}_{}_throughput'.format(precision, model_action)
         # The unit of step time is millisecond, use it to calculate the throughput with the unit samples/sec.
         millisecond_per_second = 1000
         throughput = [millisecond_per_second / step_time * self._args.batch_size for step_time in step_times]
-        metric = '{}_{}_throughput'.format(precision, model_action)
-        reduce_type = ReduceType.MIN if model_action is ModelAction.TRAIN else None
-        self._result.add_raw_data(metric, throughput)
-        self._result.add_result(metric, statistics.mean(throughput), reduce_type=reduce_type)
-        if model_action == ModelAction.INFERENCE:
-            self._process_percentile_result(metric, throughput, reduce_type=reduce_type)
+        self._result.add_raw_data(metric_s, step_times)
+        self._result.add_raw_data(metric_t, throughput)
+
+        if model_action == ModelAction.TRAIN:
+            if not self._sync_result(step_times):
+                return False
+            if self._local_rank is None or self._local_rank == 0:
+                self._result.add_result(metric_s, statistics.mean(step_times))
+                throughput = [millisecond_per_second / step_time * self._args.batch_size for step_time in step_times]
```

Comment:
Why do this? it has been calculated before in line 402

Reply:
Why are they different? I think it should be the same.

Reply:
every worker has their own raw data, but only rank0 has the summarized result.

Reply:
Sure. Let's keep it!